hazelcast/src/main/java/com/hazelcast/spi/impl/merge/AbstractNonPartitionedContainerCollector.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.merge;
+
+import com.hazelcast.spi.NodeEngine;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Base implementation of {@link AbstractContainerCollector} for non-partitioned data structures.
+ *
+ * @param <C> container of the data structure
+ */
+public abstract class AbstractNonPartitionedContainerCollector<C> extends AbstractContainerCollector<C> {
+
+    protected final Map<String, C> containers;
```

Comment:
We should probably enforce this to be `ConcurrentMap` since we know it will be accessed from different threads.